### ISSUE TITLE
Adding members shows users already belonging to the application in the list

### ISSUE THREAD

**Issue by hukka** [CSCfi/rems#1412]
Better not to even show those users who are already in, than let the user try to add them and then show an error.

**Comment by tjb** [comment]
hey everyone! i am new to learning clojure & want to participate in Hacktoberfest. is it ok if i take on this issue?

**Comment by Macroz** [comment]
Yes definitely!

**Comment by opqdonut** [comment]
@tjb Do you need help getting started? `README.md` and `docs/development.md` are good places to start. Ping us if you're stuck with something.

**Comment by tjb** [comment]
@opqdonut honestly the documentation is *really* good! i was able to get the app running very quickly. 

i am having a little issue of making the REPL work with both the backend and frontend using [Calva](https://marketplace.visualstudio.com/items?itemName=betterthantomorrow.calva) but that is ok.

I believe what will work best for this situation is create a new endpoint that the dropdown fetches from that will filter out any applicants based on application id. this allows us to make the backend the "source of truth" so to speak. I will have additional time to work on this this weekend and hopefully get a PR up by Sunday.  

an additional question i am not sure where to ask or if it is ok to log an issue for: i believe the endpoint should be paginated as well. with the test data, this is an extremely large list of users that takes a while for a return response and then populating the UI. i think that may be out of scope for this and would love to make an issue around it if that seems appropriate to everyone!

**Comment by Macroz** [comment]
The existing members are also available in the app db under `:rems.application/application :data :application/members`. So we could filter the dropdown also in the UI? This would probably be the simplest solution to the original issue.

As far as the second question goes. It's fine to file an idea issue about pagination! 

So far we are expecting (and have perf test data for) about 1000 items, also users, so we have not considered doing pagination in the backend because the amount is rather small. However now that I'm trying it out the performance is quite poor and I notice at least two things. It is transfering 92kB of data which sounds ok (roughly 100 bytes per user including name and email at least). The second and real issue is probably the rendering of the 1000 items into the `react-select` dropdown. Maybe it has poorer performance than it should. Could be that it causes also unnecessary re-rendering of the page. Good ideas about that are always also welcome!

Lastly, perhaps doing the filtering in the backend would optimize here. We use a different approach i.e. a Lucene index for searching applications on the Actions page and it's also potentially something to use for the backend of a dropdown.

**Comment by hukka** [comment]
> The second and real issue is probably the rendering of the 1000 items into the react-select dropdown.

Agreed. Though we didn't test it when I had made this issue, I have a strong hunch that the back and forth latencies caused by pagination or running the query on the backend would exceed the delay caused by loading the expected number of users. The real issue is trying to render them.

The frontend should be a bit more complex, and render something like "… and 980 more users" as the last element of the list. Filtering the 100kB of data in the frontend should happen faster than a blink of an eye. But that's still simpler and less code than pagination logic.

**Comment by Macroz** [comment]
Anything we can help you with here @tjb?

**Comment by Macroz** [comment]
@tjb do you want to continue on this issue or shall we finish it?

**Comment by tjb** [comment]
@Macroz Apologies for my radio silence on this issue. During that time life sorta happened and got swept up in things. If the team decides to take this back out of the backlog I can re-look at it!

**Comment by Macroz** [comment]
No worries @tjb! This is not high priority for us so we won't implement it for a while, and I guess it'll be up for grabs in the next Hacktoberfest.
